libcnb-cargo/tests/test.rs

```diff
@@ -1,307 +1,330 @@
-            &self.target_triple,
-        )
-    }
+    env::temp_dir()
+        // libcnb-cargo itself will canonicalize paths too.
+        // If we don't, the string representations of the path won't match. Doing this here will
+        // simplify code at the call-sites since they don't have to canonicalize themselves.
+        // Ref: https://github.com/rust-lang/rust/issues/99608
+        .canonicalize()
+        .and_then(tempdir_in)
+        .and_then(|temp_dir| copy_dir_recursively(&fixture_dir, temp_dir.path()).map(|_| temp_dir))
+}
 
-    fn run_libcnb_package(&self) -> TestOutput {
-        self.run_libcnb_package_from(".")
-    }
+fn copy_dir_recursively(source: &Path, destination: &Path) -> std::io::Result<()> {
+    match fs::create_dir(destination) {
+        Err(io_error) if io_error.kind() == ErrorKind::AlreadyExists => Ok(()),
+        other => other,
+    }?;
 
-    fn run_libcnb_package_from(&self, from_dir: &str) -> TestOutput {
-        // borrowed from assert_cmd
-        let suffix = env::consts::EXE_SUFFIX;
-
-        let target_dir = env::current_exe()
-            .ok()
-            .map(|mut path| {
-                path.pop();
-                if path.ends_with("deps") {
-                    path.pop();
-                }
-                path
-            })
-            .unwrap();
-
-        let name = "cargo-libcnb";
-
-        let cargo_libcnb = env::var_os(format!("CARGO_BIN_EXE_{name}"))
-            .map(|p| p.into())
-            .unwrap_or_else(|| target_dir.join(format!("{name}{suffix}")));
-
-        let mut cmd = Command::new(cargo_libcnb)
-            .args(["libcnb", "package", "--release"])
-            .current_dir(self.dir().join(from_dir))
-            .stdout(Stdio::piped())
-            .stderr(Stdio::piped())
-            .spawn()
-            .unwrap();
-
-        let status = cmd.wait().unwrap();
-
-        let mut stdout = String::new();
-        cmd.stdout
-            .take()
-            .unwrap()
-            .read_to_string(&mut stdout)
-            .unwrap();
-        println!("STDOUT:\n{stdout}");
+    for entry in fs::read_dir(source)? {
+        let dir_entry = entry?;
 
-        let mut stderr = String::new();
-        cmd.stderr
-            .take()
-            .unwrap()
-            .read_to_string(&mut stderr)
-            .unwrap();
-        println!("STDERR:\n{stderr}");
-
-        TestOutput {
-            stdout: String::from_utf8_lossy(stdout.as_bytes()).to_string(),
-            stderr: String::from_utf8_lossy(stderr.as_bytes()).to_string(),
-            code: status.code(),
+        if dir_entry.file_type()?.is_dir() {
+            copy_dir_recursively(&dir_entry.path(), &destination.join(dir_entry.file_name()))?;
+        } else {
+            fs::copy(dir_entry.path(), destination.join(dir_entry.file_name()))?;
         }
     }
+
+    Ok(())
 }
 
 const X86_64_UNKNOWN_LINUX_MUSL: &str = "x86_64-unknown-linux-musl";
+const TARGET_DIR_NAME: &str = "target";
```

Comment:
This is temporary, #583 will change this again (and is a fast follow-up after this PR)